allow ghc-prim 0.13, ghc 9.12

Hopefully an easy fix or hackage revision: allowing ghc-prim 0.13 will make it easier to use encoding with ghc 9.12.

Also, there are two build warnings with ghc 9.12:

```
encoding             > [ 6 of 67] Compiling Data.Encoding.BootString
encoding             > /private/var/folders/r7/f9j9c2zd1k97v47cr84j_qvc0000gp/T/stack-46111a7a03cd017e/encoding-0.10.1/Data/Encoding/BootString.hs:75:19: warning: [GHC-63394] [-Wx-partial]
encoding             >     In the use of ‘head’
encoding             >     (imported from Prelude, but defined in GHC.Internal.List):
encoding             >     "This is a partial function, it throws an error on empty lists. Use pattern matching, 'Data.List.uncons' or 'Data.Maybe.listToMaybe' instead. Consider refactoring to use "Data.List.NonEmpty"."
encoding             >    |
encoding             > 75 |         (rd,rk) = head
encoding             >    |                   ^^^^

encoding             > [67 of 67] Compiling System.IO.Encoding
encoding             > /private/var/folders/r7/f9j9c2zd1k97v47cr84j_qvc0000gp/T/stack-46111a7a03cd017e/encoding-0.10.1/System/IO/Encoding.hs:124:11: warning: [GHC-68441] [-Wdeprecations]
encoding             >     In the use of ‘hGetLine’ (imported from Data.ByteString):
encoding             >     Deprecated: "Deprecated since @bytestring-0.12@. Use 'Data.ByteString.Char8.hGetLine' instead. (Functions that rely on ASCII encodings belong in "Data.ByteString.Char8")"
encoding             >     |
encoding             > 124 |   line <- BS.hGetLine h
encoding             >     |           ^^^^^^^^^^^
```

--- a/encoding.cabal
+++ b/encoding.cabal
@@ -34,7 +34,7 @@ Library
                  bytestring >=0.9 && <0.13,
                  containers >=0.4 && <0.8,
                  extensible-exceptions >=0.1 && <0.2,
-                 ghc-prim >=0.3 && <0.13,
+                 ghc-prim >=0.3 && <0.14,
                  mtl >=2.0 && <2.4,
                  regex-compat >=0.71 && <0.96
 
@@ -129,7 +129,7 @@ executable encoding-exe
                      , filepath
                      , containers
                      , HaXml >=1.22 && <1.27
-                     , Cabal >= 2.0 && < 3.14
+                     , Cabal >= 2.0 && < 3.15
                      , encoding
   else
     buildable: False

